.github/scripts/codecheck.py

```diff
@@ -67,6 +67,23 @@ def check_crate_version_unique(crate_name):
 
     return len(versions) == 1
 
+# Ensure that the versions in the workspace's Cargo.toml are consistent
+def check_workspace_and_package_versions_equal():
+    print("==== Ensuring workspace and package versions are equal in the workspace's Cargo.toml")
+    root = toml.load('Cargo.toml')
+
+    workspace_version = root['package']['version']
+    package_version = root['workspace']['package']['version']
```

Comment:
Will this throw an exception of these keys are not defined? Maybe it could be handled explicitly so it does not prevent subsequent checks being run.

Reply:
Yes, it will. I tested it yesterday. But I don't think it's worth it. This should never happen + debugging it will tell you exactly what's going on (key isn't in dict).